app/app/src/main/java/org/omecproject/up4/impl/Up4DeviceManager.java

```diff
@@ -435,16 +435,24 @@ private void ensureInterfacesInstalled() {
     public Collection<UpfInterface> configInterfaces() {
         Collection<UpfInterface> interfaces = new ArrayList<>();
         if (config.n3Address().isPresent()) {
-            interfaces.add(UpfInterface.createN3From(config.n3Address().get(), SLICE_MOBILE));
+            if (config.sliceId().isPresent()) {
+                interfaces.add(UpfInterface.createN3From(config.n3Address().get(), config.sliceId().get()));
+            } else {
+                log.error("Missing slice ID in the netcfg to configure N3 interface for {}", config.n3Address().get());
+            }
         }
         for (Ip4Prefix uePool : config.uePools()) {
-            interfaces.add(UpfInterface.createUePoolFrom(uePool, SLICE_MOBILE));
+            if (config.sliceId().isPresent()) {
+                interfaces.add(UpfInterface.createUePoolFrom(uePool, config.sliceId().get()));
+            } else {
+                log.error("Missing slice ID in the netcfg to configure N6 interface for UE subnet {}", uePool);
+            }
         }
         Ip4Address dbufDrainAddr = config.dbufDrainAddr();
         if (dbufDrainAddr != null) {
             // TODO: change slice ID when adding multi-slice support,
```

Comment:
does this still apply ?

Reply:
yes, I would leave that comment. When adding multi-slice support, we should evaluate if still sending traffic to the default slice or using a different approach.